### PR TITLE
doc: release-notes-3.7: Update DFU/MCUmgr release notes

### PR DESCRIPTION
**Pull request by de-nordic**
Notes on additions and deprecations.

### CODE REVIEW

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -119,6 +119,9 @@ Removed APIs in this release
  * Removed ``pm_device_state_lock``, ``pm_device_state_is_locked`` and ``pm_device_state_unlock``
    functions from the Device PM APIs.
 
+ * Removed deprecated MCUmgr transport API functions have been removed: ``zephyr_smp_rx_req``,
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:122]:
```suggestion
 * Removed deprecated MCUmgr transport API functions: ``zephyr_smp_rx_req``,
```

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -1363,6 +1366,14 @@ Libraries / Subsystems
     * Fixed an issue with the SMP structure not being packed which would cause a fault on devices
       that do not support unaligned memory accesses.
 
+    * Added :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT_DYNAMIC_SVC_REGISTRATION` that allows users
+      to select whether MCUmgr BT service is statically registered at compile time or
+      dynamically at run time.
+
+    * In FS group, the TinyCrypt used for SHA calculation has been replaced with PSA calls.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:1373]:
```suggestion
    * In FS group, TinyCrypt has been replaced with PSA calls for SHA calculation.
```
maybe better?

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -1363,6 +1366,14 @@ Libraries / Subsystems
...
+
+    * Following deprecated functions have been removed
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:1375]: Cut sentence? :)

**Review by aescolar — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.7.rst (hunk 3 above)
Comment (de-nordic) [reply doc/releases/release-notes-3.7.rst:1375]: Ha. No I have decided to move it in the middle of writing the sentence.

Comment (de-nordic) [reply doc/releases/release-notes-3.7.rst]: Removed

**Review by aescolar — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**